File: docs/batch-upload/batchupload/batch.php
<?php

require_once('botr_php/api.php');

$jwplatform = new BotrAPI('API_KEY', 'API_SECRET');

// you can hardcode a file path
$csv_filepath = 'test.csv'; 

// or upload a CSV through a form 
// $csv_filepath = $_FILES['file']['tmp_name'];

if(!empty($csv_filepath) && file_exists($csv_filepath)) {
    $csv_rows = array_map('str_getcsv', file($csv_filepath));    
    $fields = array_shift($csv_rows); //parse the header line

    foreach($csv_rows AS $idx => $video) {   
        $video_meta = array_combine($fields, $video);                
        $upload_params = array();        
        
        if(!empty($video_meta['download_url'])) $upload_params['download_url'] = $video_meta['download_url'];
        if(!empty($video_meta['title'])) $upload_params['title']               = html_entity_decode($video_meta['title']);
        if(!empty($video_meta['tags'])) $upload_params['tags']                 = html_entity_decode($video_meta['tags']);
        if(!empty($video_meta['description'])) $upload_params['description']   = html_entity_decode($video_meta['description']);                    
        if(!empty($video_meta['custom'])) $upload_params['custom']             = html_entity_decode($video_meta['custom']);
        if(!empty($video_meta['date'])) $upload_params['date']                 = is_int($video_meta['date']) ? $video_meta['date'] : strtotime($video_meta['date']);

        foreach($video_meta AS $field => $val) {            
            if(strpos($field, 'custom.') !== FALSE) {
                $upload_params[$field] = $val;
            }
        }    

        $response = $jwplatform->call("/videos/create", $upload_params);
        if($response['status'] === 'ok') {
            echo $upload_params['title'] . " upload success. key = " . $response['video']['key'] . "\n";
        }
        else {
            echo $upload_params['title'] . " upload failed.\n"; 
        }
    }        
}

?>

File: docs/batch-upload/batchupload/batch.py
import csv
import HTMLParser
import time
import re
from botr_py import api

h = HTMLParser.HTMLParser()
platform = api.API('API_KEY', 'API_SECRET')

def parse_row(video={}):
	parsed_video = dict()
	for k, v in video.iteritems():
		# empty values will break the upload call
		if v is None or v == '': 
			continue
		elif k == 'download_url':
			parsed_video[k] = v
		# unescape html entities, utf8 decode values from csv
		elif k in ['title', 'description', 'tags'] or re.match('custom\.', k):
			parsed_video[k] = h.unescape(video[k].decode('utf-8'))
		# change date string to unix timestamp
		elif k == 'date':
			parsed_video[k] = video[k] if video[k].isdigit() else int(time.mktime(time.strptime(video[k], '%Y-%m-%d')))		
	return parsed_video

def upload_file(video={}):	
	# parse each parameter
	video = parse_row(video)
	# upload to platform
	return platform.call('/videos/create', video)

# main routine
with open('test.csv', 'rb') as f:
    reader = csv.reader(f)
    headings = reader.next()
    for row in reader:
    	video = dict(zip(headings, row))
        result = upload_file(video)
        # output success/failure message
        if result['status'] == 'ok':
        	print "%s upload success: key = %s." % (video['title'], result['video']['key'])
        else:
        	print "%s upload failed (%s)" % (video['title'], result['message'])